db/migrate/20250808185506_create_transactions.rb

```diff
@@ -0,0 +1,17 @@
+class CreateTransactions < ActiveRecord::Migration[8.0]
+  def change
+    create_table :transactions do |t|
+      t.references :listing, null: false, foreign_key: true
+      t.bigint :lender_id
+      t.bigint :borrower_id
```

Comment:
These can still be written as `references` like so
```suggestion
      t.references :lender, null: false, foreign_key: { to_table: :users }
      t.references :borrower, null: false, foreign_key: { to_table: :users }
```

Then with this change you don't need to explicitly create the foreign key below.